### DISCUSSION TITLE
Allow kata authors to write editorials

### DISCUSSION THREAD

**Discussion by Wrinklest** [codewars/codewars.com#2595] [Ideas]
In one of my katas, called Gravity Flip, it appears to me that even after looking at the answers, many people can't figure out why the solutions solve the problem. In many other online coding platforms, after competitions or finishing a problem, there's not only a solution but an explanation as to why the solution is what it is. Although some authors do put an explanation in the form of comments in their solution, the author's original solution is often buried under scores of "Best Practices" and "Clever" solutions. I feel that explanations and not just solutions deserve more of a spotlight after a user finishes or forfeits a problem. Especially for the more complex problems, I would personally benefit more and learn faster by reading a description of a solution algorithm, rather than trying to decipher what another user's code is supposed to do and coming up with my own hypotheses of why the solution answers the question. This could be an optional feature to kata creators that feel like their solution isn't so straightforward. What do you guys think?

**Comment by JohanWiltink** [comment]
In practice, I solve this problem by annotating my example solution as an author, and then never updating it when the kata is updated, so it can be found by sorting solutions on `Oldest`.

If I want to improve my solution, I'll solve the kata in the regular way or fork my example solution, so the annotated solution stays at the bottom, and findable.

This breaks if requirements are updated and the example solution _needs_ to be updated.

This might tie in nicely with a recent idea to annotate kata _tasks_ for solvers and translators ( and approvers of translations ), eg. for performance requirements. That information should also only be visible _after_ solving, so it might as well include an annotated example solution.

All this won't help people to understand other people's solutions that lack annotations and may not use the same cleverness the author's solution uses.

Commenting on a solution you don't understand, simply asking for an explanation, is also a possibility. I see this happening, and working, sometimes.

Between being able to find the author's example solution _most_ of the time, and being able to ask for explanations in comments on a solution or on Discord, I would not assign this any form of priority.

Other opinions are doubtlessly available.

**Comment by EloiseRosen** [comment]
I tend to leave comments in my code to explain the tricky parts. Between that and comments under solutions, there's already plenty of opportunity for anyone to explain what they're doing if they want to. I think that if someone's not explaining themselves, the issue isn't the lack of a feature, but rather lack of interest in doing so. I agree that codewars would be better if more people commented their code, but people are here for their own reasons, and some of them just want to solve katas quickly, and that's ok too. 

I like Johan's suggestion to leave a comment if something doesn't make sense to you. You can also always ask in discord and people will be happy to help!

**Comment by ejini6969** [comment]
This should be at top priority alongside with another discussion on #2588 as it adds more clarity to the task, but IMO this should not just be limited to authors but to a selected group of people which qualify and providing precise yet complete algorithm description.
